### PR TITLE
Update Wiki Docs for Select Menus

### PR DESCRIPTION
**Pull request by CheesyGamer77**
This PR updates documentation relating to Select Menus to include the [new select menu types](https://discord.com/developers/docs/change-log#new-select-menu-components).

### CODE REVIEW

File: docs/using-jda/interactions.md [hunk 1]
```diff
@@ -483,14 +492,78 @@ When a user selects their options from a dropdown and submits their choices, you
...
+                    event.reply("Choose the user to high-five")
+                        .addActionRow(selectMenu)
+                        .queue()
+                }
+            }
+        
+            override fun onEntitySelectInteraction(event: EntitySelectInteractionEvent) {
+                if (event.componentId == "choose-user") {
+                    val users = event.getMentions().getUsers();
+                    event.reply("You high-fived " + users.get(0).getAsMention()).queue()
```
Comment (freya022) [inline docs/using-jda/interactions.md:561]:
```suggestion
                    val users = event.mentions.users;
                    event.reply("You high-fived " + users.first().asMention).queue()
```

File: docs/using-jda/interactions.md [hunk 2]
```diff
@@ -483,14 +512,61 @@ When a user selects their options from a dropdown and submits their choices, you
...
+                if (event.getComponentId().equals("choose-user")) {
+                    List<User> users = event.getMentions().getUsers();
+                    event.reply("You high-fived " + users.get(0).getAsMention()).queue();
+                }
+            }
+        }
+        ```
+    === "Kotlin"
+        ```kotlin
+        object DropdownBot : ListenerAdapter() {
```
Comment (freya022) [inline docs/using-jda/interactions.md:549]:
```suggestion
        object EntityDropdownBot : ListenerAdapter() {
```

File: docs/using-jda/interactions.md [hunk 3]
```diff
@@ -469,7 +498,7 @@ When a user selects their options from a dropdown and submits their choices, you
         object DropdownBot : ListenerAdapter() {
```
Comment (freya022) [inline docs/using-jda/interactions.md:498]:
```suggestion
        object StringDropdownBot : ListenerAdapter() {
```

File: docs/using-jda/interactions.md [hunk 4]
```diff
@@ -423,29 +423,58 @@ Each non-link button requires such an ID in order to be used.
...
-When a user selects their options from a dropdown and submits their choices, you will receive a `SelectMenuInteractionEvent` for the respective interaction with the selected values.
+Entity Select Menus do not support custom choices. Instead, they derive their choices from mentionable Discord entities
+such as `User`s, `Channel`s, etc.
 
-!!! example
+![Example Entity Select Menu for Roles](../assets/images/entity_select_example.png)
+
+You can specify which entity types you wish to appear as choices by specifying the `SelectTarget`s in `EntitySelectMenu.create`.
+
+You can specify which channel types you wish to appear as channel choices by using the `setChannelTypes` method on `EntitySelectMenu.Builder`.
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:453]:
```suggestion
You can limit channel selections to specific channel types by using the `setChannelTypes` method on `EntitySelectMenu.Builder`.
```

File: docs/using-jda/interactions.md [hunk 5]
```diff
@@ -423,29 +423,58 @@ Each non-link button requires such an ID in order to be used.
...
+#### Entity Select Menus
 
-When a user selects their options from a dropdown and submits their choices, you will receive a `SelectMenuInteractionEvent` for the respective interaction with the selected values.
+Entity Select Menus do not support custom choices. Instead, they derive their choices from mentionable Discord entities
+such as `User`s, `Channel`s, etc.
 
-!!! example
+![Example Entity Select Menu for Roles](../assets/images/entity_select_example.png)
+
+You can specify which entity types you wish to appear as choices by specifying the `SelectTarget`s in `EntitySelectMenu.create`.
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:451]: Should this note that you cannot combine channel with role/user types?

File: docs/using-jda/interactions.md [hunk 6]
```diff
@@ -423,29 +423,58 @@ Each non-link button requires such an ID in order to be used.
...
+
+You can specify which channel types you wish to appear as channel choices by using the `setChannelTypes` method on `EntitySelectMenu.Builder`.
+
+!!! info
+
+    You can only specify the entity and channel types to be shown as entity select options. Further customization is not supported.
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:457]: It's not really clear what this is trying to say. Is this about not being able to limit the selection to specific users/roles? I think this needs to be made more clear than just "customization".

```suggestion
    These Select Menus do not support setting a custom list of options or customizing the appearance in any way. They always show the complete list of possible entities for the specified types.
```

Thoughts?

File: docs/using-jda/interactions.md [hunk 7]
```diff
@@ -423,29 +423,58 @@ Each non-link button requires such an ID in order to be used.
...
+
+#### Handling Select Menus
+
+When a user selects their options from a dropdown and submits their choices, you will receive either one of the following for the respective interaction:
+
+- A `StringSelectInteractionEvent` for a String Select Menu interaction
+- An `EntitySelectInteractionEvent` for an Entity Select Menu interaction
+
+Both interaction events will contain the values that were selected by the user.
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:466]:
```suggestion
Both interaction events provide the values that were selected by the user.
```

File: docs/using-jda/interactions.md [hunk 8]
```diff
@@ -483,14 +512,61 @@ When a user selects their options from a dropdown and submits their choices, you
...
+                            EntitySelectMenu.create("choose-user", SelectTarget.USER)
+                            .build())
+                        .queue();
+                }
+            }
+
+            @Override
+            public void onEntitySelectInteraction(EntitySelectInteractionEvent event) {
+                if (event.getComponentId().equals("choose-user")) {
+                    List<User> users = event.getMentions().getUsers();
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:541]:
```suggestion
                    // Mentions provide the selected values using familiar getters
                    List<User> users = event.getMentions().getUsers();
```

File: docs/using-jda/interactions.md [hunk 9]
```diff
@@ -483,14 +512,61 @@ When a user selects their options from a dropdown and submits their choices, you
...
+                    
+                    event.reply("Choose the user to high-five")
+                        .addActionRow(selectMenu)
+                        .queue()
+                }
+            }
+        
+            override fun onEntitySelectInteraction(event: EntitySelectInteractionEvent) {
+                if (event.componentId == "choose-user") {
+                    val users = event.mentions.users;
```
Comment (MinnDevelopment) [inline docs/using-jda/interactions.md:563]:
```suggestion
                    // Mentions provide the selected values using familiar getters
                    val users = event.mentions.users;
```

File: docs/using-jda/interactions.md (hunk 5 above)
Comment (CheesyGamer77) [reply docs/using-jda/interactions.md:451]: I think it'd be a good idea to mention said fact

File: docs/using-jda/interactions.md (hunk 6 above)
Comment (CheesyGamer77) [reply docs/using-jda/interactions.md:457]: The wording I initially provided is admittingly a bit _eh_. I like the wording you provided a lot better

**Review by MinnDevelopment — APPROVED** [review]

**Pull request merged**